src/buf.c

```diff
@@ -266,4 +285,10 @@ struct Buf *buf_append (struct Buf *buf, const void *ptr, int n_elem)
 	return buf;
 }
 
+struct Buf *buf_append_alloc (struct Buf *buf, const void *ptr, int n_elem)
+{
+	buf_append(&alloc_buf, ptr, n_elem);
```

Comment:
(Continued from above)
So if the value is from a strdup(), calling buf_append_alloc adds the char* (NOTE: the pointer, not the string contents) both to the requested `Buf` _and_ to `alloc_buf`.  When we are done with the `Buf`s, `buf_destroy_alloc` will free `alloc_buf` _and each_ of its entries, i.e. the exact set of pointers that were strdup()ed for all the other `Buf`s.